File: qwen2.5-coder/sistemaDeTipos/tipos.rs
fn calcular_promedio_superiores(precios: &[f64], umbral: f64) -> f64 {
    let mut suma = 0.0;
    let mut contador = 0;

    for &precio in precios.iter() {
        if precio > umbral {
            suma += precio;
            contador += 1;
        }
    }

    if contador == 0 {
        return 0.0;
    }

    suma / contador as f64
}

File: Free_ChatGPT/sistemaDeTiposRevisado/tipos.js
function promedioMayores(precios, umbral) {
    let suma = 0;
    let cantidad = 0;

    const umbralNum = Number(umbral);

    for (const precio of precios) {
        const valor = Number(precio);

        if (valor > umbralNum) {
            suma += valor;
            cantidad++;
        }
    }

    return cantidad === 0 ? 0 : suma / cantidad;
}

console.log(promedioMayores([10, '2', 30.0, "40", 50], 25));